test/api_test.js

```diff
@@ -569,6 +622,105 @@ describe('Api', () => {
     });
   });
 
+  describe('credentials', () => {
+    const P = f => path.resolve(__dirname, f);
+
+    const loadFile = (filename) => {
+      return fs.readFileSync(P(filename)).toString('base64');
+    };
+
+    const captureError = async(fn, args) => {
+      try {
+        await fn.apply(client, args);
+        return 'Should fail';
+      } catch (err) {
+        return err.code;
+      }
+    };
+
+    const expectedError = 'AuthorizationFailed';
+    const createTemporaryCredentials = taskcluster.createTemporaryCredentials;
+
+    before(() => {
+      taskcluster.createTemporaryCredentials = sandbox.stub();
+      taskcluster.createTemporaryCredentials.returns({
+        clientId: 'myClientId',
+        accessToken: 'myAccessToken',
+        certificate: 'myCertificate',
+      });
+    });
+
+    after(() => {
+      taskcluster.createTemporaryCredentials = createTemporaryCredentials;
```

Comment:
We shouldn't need to do this, rather we should be using the sandbox's `.restore()` functionality